src/models/branches/lines.jl

```diff
@@ -3,38 +3,52 @@ const From_To_Float = NamedTuple{(:from, :to),Tuple{Float64,Float64}}
 const From_To_Bus =  NamedTuple{(:from, :to),Tuple{Bus,Bus}}
 const FromTo_ToFrom_Float = NamedTuple{(:from_to, :to_from),Tuple{Float64,Float64}}
 
-
 struct Line <: Branch
     name::String
     available::Bool
     connectionpoints::From_To_Bus
     r::Float64 #[pu]
     x::Float64 #[pu]
     b::From_To_Float # [pu]
-    rate::FromTo_ToFrom_Float #MW
+    rate::Float64
     anglelimits::Min_Max #Degrees
 end
 
 function Line(name::String, available::Bool, connectionpoints::From_To_Bus,
-              r::Float64, x::Float64, b::From_To_Float, rate::FromTo_ToFrom_Float, anglelimits::Float64)
-        anglelimits = (min = -anglelimits, max = anglelimits)
-        return Line(name, available, connectionpoints, r, x, b, rate, anglelimits)
+    r::Float64, x::Float64, b::From_To_Float, rate::Float64, anglelimits::Float64)
+anglelimits = (min = -anglelimits, max = anglelimits)
+return Line(name, available, connectionpoints, r, x, b, rate, anglelimits)
 end
 
-function Line(name::String, available::Bool, connectionpoints::From_To_Bus,
-              r::Float64, x::Float64, b::From_To_Float, rate::Float64, anglelimits::Min_Max)
-        rate =  (from_to = rate, to_from = rate)
-        return Line(name, available, connectionpoints, r, x, b, rate, anglelimits)
+Line(; name = "init",
+       available = false,
+       connectionpoints = (from = Bus(), to = Bus()),
+       r = 0.0,
+       x = 0.0,
+       b = (from=0.0, to =0.0),
+       rate = 0.0,
+       anglelimits = (min = -1.57, max = 1.57)
+    ) = Line(name, available, connectionpoints, r, x, b, rate, anglelimits)
+
+struct ConstrainedLine <: Branch
+    name::String
+    available::Bool
+    connectionpoints::From_To_Bus
+    r::Float64 #[pu]
+    x::Float64 #[pu]
+    b::From_To_Float # [pu]
+    flowlimits::FromTo_ToFrom_Float #MW
+    rate::Float64
+    anglelimits::Min_Max #Degrees
 end
 
-function Line(name::String, available::Bool, connectionpoints::From_To_Bus,
```

Comment:
I like the addition of the `ConstrainedLine`. Is the constraint only with respect to the flow limit? Or is this also to w.r.t other constraints? If it's just flow limits, we should call it `MonitoredLine` as that's more in line with the industry terminology. Also, we should create an `UnMonitoredLine`.

Reply:
The constraint is only w.r.t to flow in this form. It can be extended to other limits but I think those can be generalized to flow in most cases. I will change the name to `Monitored` but I don't see why we would need the `Unmonitored`

Reply:
You can disregard the `unmonitored` part of my comment. On further thought, there are better ways to filter the enforcement of flow constraints